test/elixir_analyzer/test_suite/rpg_character_sheet_test.exs

```diff
@@ -0,0 +1,234 @@
+defmodule ElixirAnalyzer.TestSuite.RpgCharacterSheetTest do
+  use ElixirAnalyzer.ExerciseTestCase,
+    exercise_test_module: ElixirAnalyzer.TestSuite.RpgCharacterSheet
+
+  alias ElixirAnalyzer.Constants
+
+  test_exercise_analysis "example solution",
+    comments: [] do
+    defmodule RPG.CharacterSheet do
+      def welcome() do
+        IO.puts("Welcome! Let's fill out your character sheet together.")
+      end
+
+      def ask_name() do
+        name = IO.gets("What is your character's name?\n")
+        String.trim(name)
+      end
+
+      def ask_class() do
+        name = IO.gets("What is your character's class?\n")
+        String.trim(name)
+      end
+
+      def ask_level() do
+        level = IO.gets("What is your character's level?\n")
+        level = String.trim(level)
+        String.to_integer(level)
+      end
+
+      def run() do
+        welcome()
+        name = ask_name()
+        class = ask_class()
+        level = ask_level()
+
+        character = %{
+          name: name,
+          class: class,
+          level: level
+        }
+
+        IO.inspect(character, label: "Your character")
+      end
+    end
+  end
+
+  test_exercise_analysis "welcome doesn't end with IO.puts",
+    comments_include: [Constants.rpg_character_sheet_welcome_ends_with_IO_puts()] do
+    [
+      defmodule RPG.CharacterSheet do
+        def welcome() do
+          IO.puts("Welcome! Let's fill out your character sheet together.")
+          :ok
+        end
+      end,
+      defmodule RPG.CharacterSheet do
+        def welcome() do
+          text = "Welcome! Let's fill out your character sheet together."
+          IO.puts(text)
+          :ok
+        end
+      end
+    ]
+  end
```

Comment:
Since this check is about the implicit vs explicit `:ok` return, I would like to allow students to use `IO.write` instead of `IO.puts`. Could we add a test and make it pass?

The (already merged) comment can stay like it is. We don't need to customize if to say "puts or write" because it _should_ be puts, so maybe the student would get that hint from the comment too 🙂 

```suggestion
  describe "implicit :ok return" do
    test_exercise_analysis "ending with IO.write is not ideal, but allowed",
      comments_exclude: [Constants.rpg_character_sheet_welcome_ends_with_IO_puts()] do
      [
        defmodule RPG.CharacterSheet do
          def welcome() do
            IO.write("Welcome! Let's fill out your character sheet together.\n")
          end
        end
      ]
    end

    test_exercise_analysis "welcome doesn't end with IO.puts or IO write",
      comments_include: [Constants.rpg_character_sheet_welcome_ends_with_IO_puts()] do
      [
        defmodule RPG.CharacterSheet do
          def welcome() do
            IO.puts("Welcome! Let's fill out your character sheet together.")
            :ok
          end
        end,
        defmodule RPG.CharacterSheet do
          def welcome() do
            text = "Welcome! Let's fill out your character sheet together."
            IO.puts(text)
            :ok
          end
        end,
        defmodule RPG.CharacterSheet do
          def welcome() do
            text = "Welcome! Let's fill out your character sheet together."
            IO.write(text <> "\n")
            :ok
          end
        end
      ]
    end
  end
```